unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/Tests/SBC_Asserts.cs

```diff
@@ -42,34 +44,36 @@ public static IEnumerator EntitiesAreBeingCorrectlyRegistered(ParcelScene scene)
             Assert.AreEqual(0, scene.entities.Count, "entity count should be zero");
             Assert.AreEqual(0, Environment.i.world.sceneBoundsChecker.entitiesToCheckCount, "entities to check should be zero!");
         }
-        
-        public static IEnumerator EntityIsEvaluatedOnReparenting(ParcelScene scene)
+
+        public static async Task EntityIsEvaluatedOnReparenting(ParcelScene scene)
         {
             var boxShape = TestUtils.CreateEntityWithBoxShape(scene, new Vector3(8, 2, 8));
             var shapeEntity = boxShape.attachedEntities.First();
 
-            yield return null;
+            await UniTask.WaitForFixedUpdate();
             AssertMeshesAndCollidersValidState(shapeEntity.meshesInfo, true);
-         
+
             var newParentEntity = TestUtils.CreateSceneEntity(scene);
             TestUtils.SetEntityTransform(scene, newParentEntity, new DCLTransform.Model { position = new Vector3(100, 1, 100) });
-            
+
             // Our entities parenting moves the child's local position to Vector3.zero by default...
             TestUtils.SetEntityParent(scene, shapeEntity, newParentEntity);
-            
-            yield return null;
+
+            await UniTask.WaitForFixedUpdate();
             AssertMeshesAndCollidersValidState(shapeEntity.meshesInfo, false);
         }
 
-        public static IEnumerator PShapeIsInvalidatedWhenStartingOutOfBounds(ParcelScene scene)
+        public static async Task PShapeIsInvalidatedWhenStartingOutOfBounds(ParcelScene scene)
         {
             var boxShape = TestUtils.CreateEntityWithBoxShape(scene, new Vector3(20, 2, 20));
-            yield return null;
+
+            await UniTask.WaitForFixedUpdate(); // preliminary check
+            await UniTask.WaitForFixedUpdate(); // immortal process catches up
```

Comment:
I would suggest to group it in one `WaitForTwoFixedUpdates()` method, since you used it a lot

Reply:
Ah yes that's smart